src/ecs/commands/updateEnableExecuteCommandFlag.ts

```diff
@@ -29,50 +27,49 @@ export async function updateEnableExecuteCommandFlag(
         'AWS.command.ecs.runCommandInContainer.warning.disableExecuteFlag',
         'Disabling command execution will change the state of resources in your AWS account, including but not limited to stopping and restarting the service.\n Continue?'
     )
+    const alreadyEnabled = localize('AWS.command.ecs.enableEcsExec.alreadyEnabled', 'ECS Exec is already enabled for this service')
+    const alreadyDisabled = localize('AWS.command.ecs.enableEcsExec.alreadyDisabled', 'ECS Exec is already disabled for this service')
     const yes = localize('AWS.generic.response.yes', 'Yes')
     const yesDontAskAgain = localize('AWS.message.prompt.yesDontAskAgain', "Yes, and don't ask again")
     const no = localize('AWS.generic.response.no', 'No')
+    
+    const prompt = enable ? 'ecsRunCommandEnable' : 'ecsRunCommandDisable'
+    const hasExecEnabled = node.service.enableExecuteCommand
 
-    if (enable) {
-        if (hasExecEnabled) {
-            window.showInformationMessage(
-                localize('AWS.command.ecs.enableEcsExec.alreadyEnabled', 'ECS Exec is already enabled for this service')
-            )
-            return
-        }
-        if (await settings.isPromptEnabled('ecsRunCommandEnable')) {
-            const choice = await window.showWarningMessage(enableWarning, yes, yesDontAskAgain, no)
+    const warningMessage = enable ? enableWarning : disableWarning
+    const redundentActionMessage = enable ? alreadyEnabled : alreadyDisabled
+
+    if(enable === hasExecEnabled) {
+        window.showInformationMessage(redundentActionMessage)
+        return
+    }
+
+    try {
+        if (await settings.isPromptEnabled(prompt)) {
+            const choice = await window.showWarningMessage(warningMessage, yes, yesDontAskAgain, no)
             if (choice === undefined || choice === no) {
                 return
             } else if (choice === yesDontAskAgain) {
-                settings.disablePrompt('ecsRunCommandEnable')
+                settings.disablePrompt(prompt)
             }
         }
-        await node.ecs.updateService(node.service.clusterArn!, node.name, true)
-        recordEcsEnableExecuteCommand({ result: 'Succeeded', passive: false })
+        await node.ecs.updateService(node.service.clusterArn!, node.name, enable)
+        if(enable) {
+            recordEcsEnableExecuteCommand({ result: 'Succeeded', passive: false })
+        } else {
+            recordEcsDisableExecuteCommand({ result: 'Succeeded', passive: false })
+        }
```

Comment:
Add this to a finally block and just change the result (initialize that in a let outside of the try block). It cuts down on duplication and also paves a way for the block at line 42 to set a Cancelled result for the correct operation.